File: src/main/java/com/esprit/pi_project/entities/FeedBack.java
package com.esprit.pi_project.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;

@Entity
@Table(name="feedback")
@Getter
@Setter
public class FeedBack implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long idFeedback;

    private String content;
    @Enumerated(EnumType.STRING)
    @Column(name = "status")
    private StatusFeedback status;

    @ManyToOne
    @JoinColumn(name = "idEvent")
    Evenement evenement;

    @ManyToOne
    @JoinColumn(name = "id_user")
    private User User;
}

File: Frontend/src/app/interfaces/feedback.ts
import { Event } from "./event";
import { User } from "./user";

export interface Feedback {
    idFeedback?: number;
    content?: string;
    status?: StatusFeedback;
    evenement?: Event;
    user?: User;
    rating?:number;
}

export enum StatusFeedback {
    Processed,
    Unprocessed
}
